### PR TITLE
Add forms

### PR DESCRIPTION
**Pull request by Pazzo97**
### In this PR I did the following:

-  Created a method called `current_user` in ApplicationController to make current user data available to all controllers.
    -  It will return the first user from the database.
- Created forms to perform the following functions:
    - Creates a Post on behalf of the `current_user` (use the method that you created in your ApplicationController).
    - Create a comment on behalf of the `current_user` (use the method that you created in your ApplicationController).
- Allow Users to add likes to Posts.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -1,12 +1,39 @@
...
+    @post = Post.find(params[:id])
     @user = @post.author
     @comments = @post.comments
   end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
```
Comment (Danieldotcomcoder) [inline app/controllers/posts_controller.rb:17]: When I try to create a new Post , It wasn't created and I got an error `unknown attribute 'author_id' for Post`  Kindly Fix this issue  👍

**Review by Danieldotcomcoder — CHANGES_REQUESTED** [review]
Hello @Pazzo97 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
- Professional Readme File ✔✔
- Linters are setup correctly ✔✔
- Clear and nice code ✔✔

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (Pazzo97) [reply app/controllers/posts_controller.rb:17]: Hi @Danieldotcomcoder 
Thanks for your review but on my side, it is working perfectly.

Here is the screenshot I took for adding new post 

required

Comment (Danieldotcomcoder) [reply app/controllers/posts_controller.rb:17]: Hello @Pazzo97 , I tried multiple times it didn't work on my end, I'm sorry for that, I'm still trying to figure what's possibly wrong.

Comment (Pazzo97) [reply app/controllers/posts_controller.rb:17]: @Danieldotcomcoder  Can I call you on zoom? for few mins

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
+class CommentsController < ApplicationController
+  def new
+    @comment = Comment.new
+  end
+
+  def create
+    @post = Post.find(params[:post_id])
+    @comment = current_user.comments.new(post_id: @post.id, author_id: current_user.id, text: comment_params)
```
Comment (VICTOR-CHUKWU) [inline app/controllers/comments_controller.rb:8]: - [ ] there is an error about your use of author_id, your schema has user_id, so you either change all author_id to user id or change the user_id in your schema to author_id
Screenshot from 2022-08-19 19-49-43

**Review by VICTOR-CHUKWU — CHANGES_REQUESTED** [review]
Hi @Pazzo97 , 
## STATUS: CHANGES REQUIRED :recycle: 
Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: app/controllers/comments_controller.rb (hunk 2 above)
Comment (Pazzo97) [reply app/controllers/comments_controller.rb:8]: Hi @VICTOR-CHUKWU I am having an issue here, the project is running well on my side and when I push it for a review you guys get an error, would you please have a zoom call with me and help me. I am on last review and I can't change an error I am not having on this end.

Comment (VICTOR-CHUKWU) [reply app/controllers/comments_controller.rb:8]: I hope you are clear now with my explanations. nice work tho :+1:

**Review by whiz25 — APPROVED** [review]
Hi @Pazzo97, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

### Martin Afani
#### Slack: whiz25

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**